openfisca_core/holders.py

```diff
@@ -251,6 +270,71 @@ def graph(self, edges, get_input_variables_and_parameters, nodes, visited):
             return
         formula.graph_parameters(edges, get_input_variables_and_parameters, nodes, visited)
 
+    def get_memory_usage(self):
+        """
+            Gets data about the virtual memory usage of the holder.
+
+            :returns: Memory usage data
+            :rtype: dict
+
+            Exemple:
+
+            >>> holder.get_memory_usage()
+            >>> {
+            >>>    'nb_arrays': 12,  # The holder contains the variable values for 12 different periods
+            >>>    'nb_cells_by_array': 100, # There are 100 entities (e.g. persons) in our simulation
+            >>>    'cell_size': 8,  # Each value takes 8B of memory
+            >>>    'dtype': dtype('float64')  # Each value is a float 64
+            >>>    'total_nb_bytes': 10400  # The holder uses 10.4kB of virtual memory
+            >>>    }
+        """
+
+        usage = dict(
+            nb_cells_by_array = self.entity.count,
+            dtype = self.variable.dtype,
+            )
+
+        if self._array is not None:
+            # Only used when definition period is ETERNITY
```

Comment:
Should ` self._array` be called `all_period_array`, or `eternity_array` ?
Otherwise, should we amend the comment to :
"# self._array is only used when definition period is ETERNITY"

Reply:
The `_array` vs `_array_by_period` is ugly and annoying, and probably requires more than a renaming. I'll just improve the comment for this PR 🙂